src/renderer/component/userFieldNew/view.jsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { Form, FormRow, Submit } from 'component/form.js';
+
+class UserFieldNew extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      phone: '',
+      email: '',
+      country_code: '+1',
+    };
+
+    this.formatPhone = this.formatPhone.bind(this);
+  }
+
+  formatPhone(value) {
+    const { country_code } = this.state;
+    value = value.replace(/\D/g, '');
+    if (country_code === '+1') {
+      if (!value) {
+        return '';
+      } else if (value.length < 4) {
+        return value;
+      } else if (value.length < 7) {
+        return `(${value.substring(0, 3)}) ${value.substring(3)}`;
+      }
+      const fullNumber = `(${value.substring(0, 3)}) ${value.substring(3, 6)}-${value.substring(
+        6
+      )}`;
+      return fullNumber.length <= 14 ? fullNumber : fullNumber.substring(0, 14);
+    }
+    return value;
+  }
+
+  formatCountryCode(value) {
+    if (value) {
+      return `+${value.replace(/\D/g, '')}`;
+    }
+    return '+1';
+  }
+
+  handleChanged(event, fieldType) {
+    const formatter = {
+      email: _ => _,
+      phone: this.formatPhone,
+      country_code: this.formatCountryCode,
+    };
+    this.setState({
+      [fieldType]: formatter[fieldType](event.target.value),
+    });
+  }
+
+  handleSubmit() {
+    const { email, phone, country_code } = this.state;
+    if (phone) {
+      this.props.addUserPhone(phone.replace(/\D/g, ''), country_code.replace(/\D/g, ''));
+    } else {
+      this.props.addUserEmail(email);
+    }
+  }
+
+  render() {
+    const { cancelButton, emailErrorMessage, phoneErrorMessage, isPending, fieldType } = this.props;
+
+    return fieldType === 'phone' ? (
```

Comment:
Is there a particular reason we use a single component to do two (seemingly) distinct duties, as opposed to `UserEmailNew` and `UserPhoneNew` or equivalent?

Reply:
this was in the spirit of DRY a la our coding maxims.

There is less of a case for userFieldNew to be used for two field types, but userFieldVerify is basically the same for both types. I made userFieldNew to be consistent with that.

Reply:
I don't see very much savings here aside from maybe some boilerplate in index.js, and it adds complexity to declaration and handling changes. I'd split these.